src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,513 @@
+            )
+
+            # DEHB may sample 2 very similar trial, that gets the same ID because
+            # for instance the precision is low and rounding the HP values leads to
+            # 2 identical trials. It this case you will have has_suggested(new_trial)
+            # is True, and you will discard this trial.
+            # It's fine to discard the trial, but we should keep track of the job_info.
+            # DEHB does not know that we discarded the trial and will be waiting for the
+            # result.We need to keep track of both job_info so that when we have
+            # the result of the first trial, we assign it to the second job_info as well.
+
+            if not self.has_suggested(new_trial):
+                # Store metadata
+                self.job_infos[self.get_id(new_trial)].append(job_info)
+                self.register_job(job_info)
+
+                # Standard Orion
+                self.register(new_trial)
+                trials.append(new_trial)
+                logger.debug("Suggest new trials %s", new_trial)
+            else:
+                logger.debug("Already suggested %s", new_trial)
+
+                # Do we already have a result for this trial ?
+                result = self.job_results.get(self.get_id(new_trial))
+
+                # Keep track of duplicated jobs per brackets
+                # Bracket is only done after we reach the budget for unique
+                # jobs
+                self.duplicates[str(job_info["budget"])] += 1
+
+                # if so observe it right now and discard
+                if result is not None:
+                    self._dehb_observe(job_info, *result)
+                else:
+                    # else we need to keep track of it to observe it later
+                    self.job_infos[self.get_id(new_trial)].append(job_info)
+
+        return trials
+
+    def observe(self, trials: List[Trial]) -> None:
+        """Observe the `trials` new state of result.
+
+        Parameters
+        ----------
+        trials: list of ``orion.core.worker.trial.Trial``
+           Trials from a `orion.algo.space.Space`.
+
+        """
+        for trial in trials:
+            if not self.has_suggested(trial):
+                logger.debug("Ignore unseen trial %s", trial)
+                continue
+
+            if self.has_observed(trial):
+                logger.debug("Ignore already observed trial %s", trial)
+                continue
+
+            self.register(trial)
+
+            if trial.status == "completed":
+                self.observe_one(trial)
+                logger.debug(
+                    "Observe trial %s (Remaining %d)", trial, len(self.job_infos)
+                )
+
+    def observe_one(self, trial: Trial) -> None:
+        """Observe a single trial"""
+
+        # Get all the job sampled by DEHB, it might be more than one
+        job_infos = self.job_infos.get(self.get_id(trial), [])
```

Comment:
Note to self, this kind of stuff (using `self.get_id`) might get impacted by #833 . Need to double-check.

Reply:
Fixed in 412e0c8fbfcb18883d361cafa39b5292bcef6cc5. I needed to keep it because the algorithm we wrap may actually suggest identical trials within the transformed space so we still need to handle them that way. I adjusted the note about duplicates however since the source of duplicates is just for discrete dimensions and not because of transformations.